Issue517 create data page

Hurlyburly2 opened the pull request:
Adds data page as outlined in https://github.com/pod4lib/aggregator/issues/517.
Let me know if there are any differences in ways that this would've been implemented, happy to accept feedback.

<img width="1312" alt="Screen Shot 2022-04-13 at 5 56 21 PM" src="https://user-images.githubusercontent.com/46659222/163277029-1a1a4bcd-c63d-43f6-a728-d9a290e85a71.png">

ggeisler commented:
@Hurlyburly2 I'd suggest removing the Bootstrap classes on the heading elements (e.g. the `.mt-5 .mb-4` classes on `<h2 class="mt-5 mb-4">`). To ensure consistency throughout the application, we should rely on the default styling that comes with the different heading levels. If for some reason, we want to adjust them, I think it's better to do that in a SCSS file where we can apply the adjusted styling to the heading level globally, throughout the application.

I'm happy to go through the application and adjust visual design things like heading spacing later, when the application is more fully updated.

marlo-longley commented on config/routes.rb:
```diff
@@ -25,6 +25,8 @@
   get 'charts/uploads', to: 'charts#uploads', as: :uploads_chart
   get 'charts/records', to: 'charts#records', as: :records_chart
+  
+  get '/data', to: 'data#index'
```
@Hurlyburly2 this is more a question since I'm newer to rails - what is the reason not to do resourceful routing since index is a default/included route? something like `resources :data, only: [:index]`

Hurlyburly2 replied:
@marlo-longley I was unsure of the actual differences myself (my Rails is kind of rusty!) so I checked with a coworker who's much more expert. Both methods create the same route, but `resources` establishes some helper functions and corresponds to a resource. In this case, we're serving up a pretty static page, and there is no model for Data to which CRUD stuff is being done.

Hurlyburly2 commented:
@ggeisler I've made the requested class removals- it does make the page look a little smooshed together, but as long as a style pass is planned then that's no problem.
I also added some additional tests regarding page access, which revealed that only admin had access to the /data route. I've refactored so logged in users also have access to the data page. While it's behaving as desired, I'm unsure if the `skip_authorization_check` usage is correct in the Data Controller. Please keep an eye on that in particular when you next take a look.

ggeisler commented:
> I've made the requested class removals- it does make the page look a little smooshed together, but as long as a style pass is planned then that's no problem.

Sounds good @Hurlyburly2. I'll definitely be doing a thorough round of visual design updates. If the page formatting looks really bad, I'll do that sooner rather than later.

Hurlyburly2 commented on app/controllers/data_controller.rb:
```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
```
use existing pages controller

corylown commented on app/controllers/data_controller.rb:
```diff
@@ -0,0 +1,8 @@
+# frozen_string_literal: true
+
+# :nodoc:
+class DataController < ApplicationController
```
Instead of creating a new controller for this, I think using the existing `PagesController` would be appropriate because it's a static content page. Just add a new action `def data; end` to the pages controller.

corylown replied:
Authorization in the controller action could look like:

```
  def data
    authorize! :read, :pages_data
  end
```

corylown commented on app/views/data/index.html.erb:
```diff
@@ -0,0 +1,34 @@
+<div class="container index-page">
```
You would move and rename this file to `app/views/pages/data.html.erb`

corylown commented on app/views/shared/_nav.html.erb:
```diff
@@ -12,7 +12,8 @@
         <li class='nav-item'><%= link_to t('.providers'), organizations_url, class: 'nav-link' %></li>
         <% if can? :manage, :dashboard_controller %>
-        <li class='nav-item'><%= link_to t('.activity'), activity_url, class: 'nav-link', data: { turbolinks: false } %></li>
+          <li class='nav-item'><%= link_to t('.activity'), activity_url, class: 'nav-link', data: { turbolinks: false } %></li>
+          <li class='nav-item'><%= link_to t('.data'), data_url, class: 'nav-link', data: { turbolinks: false } %></li>
         <% end %>
```
Wrap the data link in its own ability condition:

```suggestion
          <li class='nav-item'><%= link_to t('.activity'), activity_url, class: 'nav-link', data: { turbolinks: false } %></li>
        <% end %>
        <% if can? :read, :pages_data %>
          <li class='nav-item'><%= link_to t('.data'), data_url, class: 'nav-link', data: { turbolinks: false } %></li>
        <% end %>
```

corylown commented on config/locales/en.yml:
```diff
@@ -57,6 +57,25 @@ en:
         contributing: Help with contributing or using POD data
         framework: Data provider and usage framework
         github: GitHub repository
+  data:
+    index:
```
```suggestion
  pages:
    data:
```

corylown commented on config/routes.rb:
```diff
@@ -25,6 +25,8 @@
   get 'charts/uploads', to: 'charts#uploads', as: :uploads_chart
   get 'charts/records', to: 'charts#records', as: :records_chart
+  
+  get '/data', to: 'data#index'
```
```suggestion
  get '/data', to: 'pages#data'
```

corylown reviewed (changes requested):
In addition to the inline comments/changes you'll need to update the ability model, something like:

```
    if user.roles.any?
      # ...
      can :read, :pages_data
    end
```
